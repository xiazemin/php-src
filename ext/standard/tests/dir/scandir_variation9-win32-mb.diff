004+   string(51) "私はガラスを食べられますfile2.tmp"
004-   string(45) "私はガラスを食べられますfile2.tmp"
006+   string(51) "私はガラスを食べられますfile1.tmp"
006-   string(45) "私はガラスを食べられますfile1.tmp"
014+   string(51) "私はガラスを食べられますfile2.tmp"
014-   string(45) "私はガラスを食べられますfile2.tmp"
016+   string(51) "私はガラスを食べられますfile1.tmp"
016-   string(45) "私はガラスを食べられますfile1.tmp"
028+   string(51) "私はガラスを食べられますfile1.tmp"
028-   string(45) "私はガラスを食べられますfile1.tmp"
030+   string(51) "私はガラスを食べられますfile2.tmp"
030-   string(45) "私はガラスを食べられますfile2.tmp"