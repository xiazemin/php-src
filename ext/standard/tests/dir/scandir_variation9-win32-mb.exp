*** Testing scandir() : usage variations ***
array(4) {
  [0]=>
  string(45) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(45) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(45) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(45) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(1) "."
  [1]=>
  string(2) ".."
  [2]=>
  string(45) "私はガラスを食べられますfile1.tmp"
  [3]=>
  string(45) "私はガラスを食べられますfile2.tmp"
}
===DONE===