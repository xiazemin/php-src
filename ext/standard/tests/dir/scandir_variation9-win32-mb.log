
---- EXPECTED OUTPUT
*** Testing scandir() : usage variations ***
array(4) {
  [0]=>
  string(45) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(45) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(45) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(45) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(1) "."
  [1]=>
  string(2) ".."
  [2]=>
  string(45) "私はガラスを食べられますfile1.tmp"
  [3]=>
  string(45) "私はガラスを食べられますfile2.tmp"
}
===DONE===
---- ACTUAL OUTPUT
*** Testing scandir() : usage variations ***
array(4) {
  [0]=>
  string(51) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(51) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(51) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(51) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(1) "."
  [1]=>
  string(2) ".."
  [2]=>
  string(51) "私はガラスを食べられますfile1.tmp"
  [3]=>
  string(51) "私はガラスを食べられますfile2.tmp"
}
===DONE===
---- FAILED
