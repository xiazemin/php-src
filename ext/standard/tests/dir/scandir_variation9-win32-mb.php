<?php
/* Prototype  : array scandir(string $dir [, int $sorting_order [, resource $context]])
 * Description: List files & directories inside the specified path 
 * Source code: ext/standard/dir.c
 */

/*
 * Pass different integers as $sorting_order argument to test how scandir()
 * re-orders the array
 */

echo "*** Testing scandir() : usage variations ***\n";

// include for create_files/delete_files functions
include(dirname(__FILE__) . '/../file/file.inc');

// create directory and files
$dir = dirname(__FILE__) . '/私はガラスを食べられますscandir_variation9';
mkdir($dir);
@create_files($dir, 2, "numeric", 0755, 1, "w", "私はガラスを食べられますfile");

// different ints to pass as $sorting_order argument
$ints = array (PHP_INT_MAX, -PHP_INT_MAX, 0);

foreach($ints as $sorting_order) {
	var_dump( scandir($dir, $sorting_order) );
}

delete_files($dir, 2, "私はガラスを食べられますfile");
?>
===DONE===
