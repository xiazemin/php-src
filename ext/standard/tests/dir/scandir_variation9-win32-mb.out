*** Testing scandir() : usage variations ***
array(4) {
  [0]=>
  string(51) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(51) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(51) "私はガラスを食べられますfile2.tmp"
  [1]=>
  string(51) "私はガラスを食べられますfile1.tmp"
  [2]=>
  string(2) ".."
  [3]=>
  string(1) "."
}
array(4) {
  [0]=>
  string(1) "."
  [1]=>
  string(2) ".."
  [2]=>
  string(51) "私はガラスを食べられますfile1.tmp"
  [3]=>
  string(51) "私はガラスを食べられますfile2.tmp"
}
===DONE===